lib/handlebars/runtime.js

```diff
@@ -67,9 +67,9 @@ export function template(templateSpec, env) {
 
   // Just add water
   let container = {
-    strict: function(obj, name) {
-      if (!(name in obj)) {
-        throw new Exception('"' + name + '" not defined in ' + obj);
+    strict: function(obj, name, loc) {
+      if (!obj || !(name in obj)) {
```

Comment:
Could you explain, why you have added this part (`!obj || ...`).

Reply:
because of this case `{{foo.bar}}` in strict when foo is unknown it will throw `cannot read property 'bar' of undefined `before it can throw the exception with source location.